magda-storage-api/src/MagdaMinioClient.ts

```diff
@@ -0,0 +1,123 @@
+import { Probe } from "@magda/typescript-common/dist/express/status";
+import ObjectFromStore from "./ObjectFromStore";
+import ObjectStoreClient from "./ObjectStoreClient";
+import { Stream, Readable } from "stream";
+
+const Minio = require("minio");
+
+export default class MagdaMinioClient implements ObjectStoreClient {
+    private readonly bucket: string;
+    private readonly client: any;
+
+    constructor({ endPoint, port, useSSL, accessKey, secretKey, bucket }: any) {
+        this.bucket = bucket;
+        this.client = new Minio.Client({
+            endPoint,
+            port,
+            useSSL,
+            accessKey,
+            secretKey
+        });
+    }
+
+    readonly statusProbe: Probe = () => {
+        return this.client.bucketExists(
+            this.bucket,
+            (err: boolean | null, exists: boolean) => {
+                if (err) {
+                    return Promise.resolve({
+                        ready: false,
+                        error: "Bucket Probe returned an error."
+                    });
+                }
+                if (exists) {
+                    return Promise.resolve({ ready: true });
+                } else {
+                    return Promise.resolve({
+                        ready: false,
+                        error: "Bucket does not exist."
+                    });
+                }
+            }
+        );
+    };
+
+    getFile(fileName: string): ObjectFromStore {
+        const self = this;
```

Comment:
This is OK but if you switch to arrow functions you don't have to do it, `this` will stay as `this` :)

Reply:
True. Switched to using arrow functions.